metadata-ingestion/src/datahub/ingestion/source/unity/config.py

```diff
@@ -304,3 +313,32 @@ def include_metastore_warning(cls, v: bool) -> bool:
             logger.warning(msg)
             add_global_warning(msg)
         return v
+
+    @pydantic.root_validator(skip_on_failure=True)
+    def set_warehouse_id_from_profiling(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        profiling: Optional[UnityCatalogProfilerConfig] = values.get("profiling")
+        if not values.get("warehouse_id") and profiling and profiling.warehouse_id:
+            values["warehouse_id"] = values["profiling"].warehouse_id
+        if (
+            values.get("warehouse_id")
+            and profiling
+            and profiling.warehouse_id
+            and values["warehouse_id"] != profiling.warehouse_id
+        ):
+            raise ValueError(
+                "When `warehouse_id` is set, it must match the `warehouse_id` in `profiling`."
+            )
+
+        if values.get("include_hive_metastore") and not values.get("warehouse_id"):
+            raise ValueError(
+                "When `include_hive_metastore` is set, `warehouse_id` must be set."
+            )
+
+        return values
+
+    @pydantic.validator("schema_pattern", always=True)
+    def schema_pattern_should__always_deny_information_schema(
+        cls, v: AllowDenyPattern
+    ) -> AllowDenyPattern:
+        v.deny.append(".*\\.information_schema")
```

Comment:
this pattern of "there's a few extra db/schemas that we always want to deny" happens in other sources too, and we've been abusing the user-facing allow/deny pattern to set "system" configs. 

When we revisit the sql common refactoring, I'd like to think about moving system-level deny patterns to a class variable instead of reusing the user-facing config